services/subscriptions/subscription.go

```diff
@@ -0,0 +1,56 @@
+package subscriptions
+
+import (
+	"fmt"
+	"time"
+)
+
+type SubscriptionID string
+
+type Subscription struct {
+	id     SubscriptionID
+	signal *filterSignal
+	quit   chan interface{}
```

Comment:
idiomatically chan struct{} should be used. because empty struct{} consumes zero bytes.